File: mayonez-base/src/main/java/mayonez/application/OperatingSystem.java
package mayonez.application;

/**
 * An operating system of a computer running Java.
 *
 * @author SlavSquatSuperstar
 */
public enum OperatingSystem {

    /**
     * An unknown or undefined operating system.
     */
    UNKNOWN("Unknown OS"),

    /**
     * The FreeBSD operating system or its derivatives.
     */
    FREE_BSD("FreeBSD"),

    /**
     * The GNU/Linux family of operating systems.
     */
    LINUX("Linux"),

    /**
     * The macOS/OS X operating system.
     */
    MAC_OS("macOS"),

    /**
     * The Microsoft Windows operating system.
     */
    WINDOWS("Windows");

    private static final OperatingSystem current;

    static {
        var osName = System.getProperty("os.name");
        if (osName.contains("bsd")) current = FREE_BSD;
        else if (osName.contains("linux")) current = LINUX;
        else if (osName.contains("mac")) current = MAC_OS;
        else if (osName.contains("windows")) current = WINDOWS;
        else current = UNKNOWN;
    }

    private final String name;

    OperatingSystem(String name) {
        this.name = name;
    }

    /**
     * Get the current operating system of this device running Java.
     *
     * @return the current OS.
     */
    public static OperatingSystem getCurrent() {
        return current;
    }

    @Override
    public String toString() {
        return name;
    }

}

File: mayonez-tools/src/main/kotlin/mayonez/util/OperatingSystem.kt
package mayonez.util

/**
 * An operating system of a computer running Java.
 *
 * @author SlavSquatSuperstar
 */
enum class OperatingSystem(private val osName: String) {

    /** The FreeBSD operating system. */
    FREE_BSD("FreeBSD"),

    /** The GNU/Linux family of operating systems. */
    LINUX("Linux"),

    /** The macOS/OS X operating system. */
    MAC_OS("macOS"),

    /** The Microsoft Windows family of operating systems. */
    WINDOWS("Windows"),

    /** An unknown or undefined operating system. */
    UNKNOWN("Unknown OS");

    override fun toString(): String = osName

    companion object {
        private lateinit var current: OperatingSystem

        /**
         * Gets the current operating system of this device running Java.
         *
         * @return the current OS.
         */
        @JvmStatic
        fun getCurrent(): OperatingSystem {
            if (!OperatingSystem::current.isInitialized) {
                current = getFromName(System.getProperty("os.name").lowercase())
            }
            return current
        }

        private fun getFromName(osName: String): OperatingSystem {
            return if (osName.contains("bsd")) {
                FREE_BSD
            } else if (osName.contains("linux")) {
                LINUX
            } else if (osName.contains("mac") || osName.contains("darwin")) {
                MAC_OS
            } else if (osName.contains("windows")) {
                WINDOWS
            } else {
                UNKNOWN
            }
        }
    }

}